packages/legacy/core/App/components/listItems/NotificationListItem.tsx

```diff
@@ -82,7 +82,11 @@ const NotificationListItem: React.FC<NotificationListItemProps> = ({
   const [declineModalVisible, setDeclineModalVisible] = useState(false)
   const [action, setAction] = useState<any>()
   const [closeAction, setCloseAction] = useState<any>()
-  const connection = useConnectionById(notification.connectionId ?? '')
+  const isNotCustomNotificationRecord =
+    notification instanceof BasicMessageRecord ||
+    notification instanceof CredentialExchangeRecord ||
+    notification instanceof ProofExchangeRecord
+  const connection = isNotCustomNotificationRecord ? useConnectionById(notification.connectionId ?? '') : undefined
```

Comment:
Hooks should not be called conditionally, it's against the "Rules of Hooks"

Reply:
@bryce-mcmath done!

Reply:
Yes, @MosCD3